src/lib/storage/vector_compression/simd_bp128/simd_bp128_decompressor.cpp

```diff
@@ -21,11 +21,48 @@ SimdBp128Decompressor::SimdBp128Decompressor(const SimdBp128Decompressor& other)
       _cached_meta_info{other._cached_meta_info},
       _cached_block{std::make_unique<std::array<uint32_t, Packing::block_size>>(*other._cached_block)} {}
 
-void SimdBp128Decompressor::_read_meta_info(size_t meta_info_offset) {
+SimdBp128Decompressor::SimdBp128Decompressor(SimdBp128Decompressor&& other) noexcept
+    : _data{other._data},
+      _size{other._size},
+      _cached_meta_info_offset{other._cached_meta_info_offset},
+      _cached_meta_block_first_index{other._cached_meta_block_first_index},
+      _cached_block_first_index{other._cached_block_first_index},
+      _cached_meta_info{other._cached_meta_info},
+      _cached_block{std::move(other._cached_block)} {}
+
+SimdBp128Decompressor& SimdBp128Decompressor::operator=(const SimdBp128Decompressor& other) {
+  if (&other != this) {
+    _data = other._data;
+    _size = other._size;
+    _cached_meta_info_offset = other._cached_meta_info_offset;
+    _cached_meta_block_first_index = other._cached_meta_block_first_index;
+    _cached_block_first_index = other._cached_block_first_index;
+    _cached_meta_info = other._cached_meta_info;
+    _cached_block = std::make_unique<std::array<uint32_t, Packing::block_size>>(*other._cached_block);
```

Comment:
Don't put another day into this, but why do we need a pointer here?

Reply:
Not sure and to be honest, I don't dare touching anything that comes close to the SIMD packing. There is a lot of pointer and alignment magic that I don't understand. :(

Reply:
ok